kernel/Kconfig

```diff
@@ -319,6 +319,13 @@ config BOOT_DELAY
 	  achieved by waiting for DCD on the serial port--however, not
 	  all serial ports have DCD.
 
+config BOOT_DELAY_RANDOM
+	bool "Randomize boot delay"
+	help
+	  This option randomizes the bootup delay between 0 and the
+	  value of BOOT_DELAY. This can be used e.g., to jitter boot-up
+	  of wireless network devices.
```

Comment:
Pedantry: this needs to specify units.  How long is "1"?   One reasonable choice IMHO is to bake it into the API and call this CONFIG_BOOT_DELAY_RANDOM_MS

Reply:
As it says, the value is between 0 and BOOT_DELAY. BOOT_DELAY is in milliseconds:
https://github.com/zephyrproject-rtos/zephyr/pull/25627/files#diff-a0dbba97dac305f5927e75a408d79a25R309-R310